common/scala/src/main/scala/whisk/common/tracing/OpenTracingProvider.scala

```diff
@@ -122,19 +122,26 @@ class OpenTracer(val tracer: Tracer, tracingConfig: TracingConfig, ticker: Ticke
     }
   }
 
-  private def clear(transactionId: TransactionId): Unit = {
+  private def clear(transactionId: TransactionId, withErrorMessage: Option[String]): Unit = {
     spanMap.get(transactionId.meta.id).foreach {
       case head :: Nil =>
+        withErrorMessage.foreach(setErrorTags(head, _))
         head.finish()
         spanMap.remove(transactionId.meta.id)
         contextMap.remove(transactionId.meta.id)
       case head :: tail =>
+        withErrorMessage.foreach(setErrorTags(head, _))
         head.finish()
         spanMap.put(transactionId.meta.id, tail)
       case Nil =>
     }
   }
 
+  private def setErrorTags(span: Span, message: => String): Unit = {
+    span.setTag("error", true)
+    span.setTag("message", message)
```

Comment:
Tag is usually added to make it easy to search for a specific span. Since key in this case is always going to be "error" do you think its more appropriate case for `span.log()` ?

Reply:
Hi @sandeep-paliwal, thanks for suggestion.
I think the big difference between tag and log is timestamp; as you mentioned, the tag is much easier to search for a specific span, or make it explicit on what's going on.

I've tried [capturing errors guidance in open tracing data model standard](
https://github.com/opentracing/specification/blob/master/semantic_conventions.md#captured-errors). This is not that useful to know that what's really going on in Jaegar. 

Some code snippet:
```Scala
private def setErrorTags(span: Span, message: => String): Unit = {
  span.log(Map("event" -> "error", "message" -> message).asJava)
}
```
Snapshot:
![1](https://user-images.githubusercontent.com/13405580/45165612-bf6f2100-b227-11e8-8f40-3fbb42c0f76d.png)
The error is not explicit.

![2](https://user-images.githubusercontent.com/13405580/45165615-c138e480-b227-11e8-87d7-6458a4391829.png)

Am I missing something? If not, I think an alternative way is still use error as tag, and attach the log that we can eliminate the message tag field, if this is necessary. WDYT?

i.e.
```Scala
private def setErrorTags(span: Span, message: => String): Unit = {
  span.setTag("error", true)
  span.log(Map("event" -> "error", "message" -> message).asJava)
}
```

Reply:
Yeah I see that just adding log to trace is not that helpful in UI. I guess using tag has a better option then. Is there any use case to search for all failed invocations with a specific error? In that case error message as tag will also be useful compared to log().